src/Language/DifferentialDatalog/Compile.hs

```diff
@@ -956,6 +977,20 @@ compileMainLib d d3log_rel_map nodes cstate =
     reexports = "pub mod typedefs" $+$
                 (braces' $ ppModuleReexports d mod_reexports)
 
+-- The 'SOURCES' static variable contains complete DDlog source code embedded
+-- inside the compiled binary.
+mkSources :: (?specname::String, ?modules::M.Map ModuleName DatalogModule) => Doc
+mkSources =
+    "static SOURCES: &'static [ddlog_profiler::SourceFile] = &[" $$
+    (nest' $ vcommaSep sources)                                  $$
+    "];"
+    where
+    sources = map (\DatalogModule{..} ->
```

Comment:
should there exist an option to generate 'stripped' binaries, without this stuff?

Reply:
Probably, but I'm inclined to make it "future work".